cmd/config/config.go

```diff
@@ -452,6 +452,29 @@ func ReadUpgradeSpec(r *v1.RunConfig, flags *pflag.FlagSet) (*v1.UpgradeSpec, er
 	return upgrade, err
 }
 
+func ReadUpgradeRecoverySpec(r *v1.RunConfig, flags *pflag.FlagSet) (*v1.UpgradeSpec, error) {
+	upgrade, err := config.NewUpgradeSpec(r.Config)
+	if err != nil {
+		return nil, fmt.Errorf("failed initializing upgrade recovery spec: %v", err)
+	}
+	vp := viper.Sub("upgrade-recovery")
```

Comment:
Are we sure this is correct? shouldn't it be `vp := viper.Sub("upgrade")`? IIRC this states the sub schema item, hence it maps whatever is loaded from the yaml under `upgrade-recovery` stanza. I guess this could be tested and validated by checking this method properly reads `upgrade` section of a yaml. I assume this falls into the scope of `cmd/config/config_test.go`.

I think this should be fixed or at least validated with a unit test in this PR.

Reply:
Yes, sorry I got confused (again :D ) with it.
I removed it all and just added a (bit ugly) flag to sanitize the same stanza when used for recovery only, since it's the only logical difference in there.